aws/resource_aws_docdb_subnet_group.go

```diff
@@ -0,0 +1,230 @@
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDocDBSubnetGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDocDBSubnetGroupCreate,
+		Read:   resourceAwsDocDBSubnetGroupRead,
+		Update: resourceAwsDocDBSubnetGroupUpdate,
+		Delete: resourceAwsDocDBSubnetGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"name_prefix"},
+				ValidateFunc:  validateDocDBSubnetGroupName,
+			},
+			"name_prefix": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"name"},
+				ValidateFunc:  validateDocDBSubnetGroupNamePrefix,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Managed by Terraform",
+			},
+
+			"subnet_ids": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsDocDBSubnetGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).docdbconn
+	tags := tagsFromMapDocDB(d.Get("tags").(map[string]interface{}))
+
+	subnetIdsSet := d.Get("subnet_ids").(*schema.Set)
+	subnetIds := make([]*string, subnetIdsSet.Len())
+	for i, subnetID := range subnetIdsSet.List() {
+		subnetIds[i] = aws.String(subnetID.(string))
+	}
+
+	var groupName string
+	if v, ok := d.GetOk("name"); ok {
+		groupName = v.(string)
+	} else if v, ok := d.GetOk("name_prefix"); ok {
+		groupName = resource.PrefixedUniqueId(v.(string))
+	} else {
+		groupName = resource.UniqueId()
+	}
+
+	createOpts := docdb.CreateDBSubnetGroupInput{
+		DBSubnetGroupName:        aws.String(groupName),
+		DBSubnetGroupDescription: aws.String(d.Get("description").(string)),
+		SubnetIds:                subnetIds,
+		Tags:                     tags,
+	}
+
+	log.Printf("[DEBUG] Create DocDB Subnet Group: %#v", createOpts)
+	_, err := conn.CreateDBSubnetGroup(&createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating DocDB Subnet Group: %s", err)
+	}
+
+	d.SetId(groupName)
+
+	return resourceAwsDocDBSubnetGroupRead(d, meta)
+}
+
+func resourceAwsDocDBSubnetGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).docdbconn
+
+	describeOpts := docdb.DescribeDBSubnetGroupsInput{
+		DBSubnetGroupName: aws.String(d.Id()),
+	}
+
+	var subnetGroups []*docdb.DBSubnetGroup
+	if err := conn.DescribeDBSubnetGroupsPages(&describeOpts, func(resp *docdb.DescribeDBSubnetGroupsOutput, lastPage bool) bool {
+		for _, v := range resp.DBSubnetGroups {
```

Comment:
I believe this can be simplified to `subnetGroups = append(subnetGroups, resp.DBSubnetGroups...)` and will be flagged shortly when we enable the `gosimple` linter 👍

Reply:
Fixed.